src/sentry/identity/oauth2.py

```diff
@@ -152,70 +154,38 @@ def get_oauth_data(self, payload):
 
         return data
 
-    def handle_refresh_error(self, req, payload):
-        error_name = "unknown_error"
-        error_description = "no description available"
-        for name_key in ["error", "Error"]:
-            if name_key in payload:
-                error_name = payload.get(name_key)
-                break
-
-        for desc_key in ["error_description", "ErrorDescription"]:
-            if desc_key in payload:
-                error_description = payload.get(desc_key)
-                break
-
-        formatted_error = f"HTTP {req.status_code} ({error_name}): {error_description}"
-
-        if req.status_code == 401:
-            logger.info(
-                "identity.oauth.refresh.identity-not-valid-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
-            )
-            raise IdentityNotValid(formatted_error)
-
-        if req.status_code == 400:
-            # this may not be common, but at the very least Google will return
-            # an invalid grant when a user is suspended
-            if error_name == "invalid_grant":
-                logger.info(
-                    "identity.oauth.refresh.identity-not-valid-error",
-                    extra={
-                        "error_name": error_name,
-                        "error_status_code": req.status_code,
-                        "error_description": error_description,
-                        "provider_key": self.key,
-                    },
-                )
-                raise IdentityNotValid(formatted_error)
-
-        if req.status_code != 200:
-            logger.info(
-                "identity.oauth.refresh.api-error",
-                extra={
-                    "error_name": error_name,
-                    "error_status_code": req.status_code,
-                    "error_description": error_description,
-                    "provider_key": self.key,
-                },
+    def get_refresh_token(
+        self, refresh_token, url: str, identity: Identity | RpcIdentity, **kwargs: Any
```

Comment:
Instead of passing an untyped kwargs object, could we add the relevant options as strongly typed optional params instead? The only one I see needed here is `verify_ssl`.

Reply:
this is difficult

VSTS passes `redirect_url`
https://github.com/getsentry/sentry/blob/471bae7f33f87256ef612a61a5ebd64832c69b49/src/sentry/integrations/vsts/client.py#L204-L211
then uses `redirect_url` in `get_refresh_token_params`
https://github.com/getsentry/sentry/blob/471bae7f33f87256ef612a61a5ebd64832c69b49/src/sentry/identity/vsts/provider.py#L79-L103

Gitlab passes `refresh_token_url` and `verify_ssl`
https://github.com/getsentry/sentry/blob/471bae7f33f87256ef612a61a5ebd64832c69b49/src/sentry/integrations/gitlab/client.py#L115-L121
uses the `refresh_token_url` instead of the one normally hardcoded to the provider
https://github.com/getsentry/sentry/blob/471bae7f33f87256ef612a61a5ebd64832c69b49/src/sentry/identity/gitlab/provider.py#L97-L114